WalletWasabi.Fluent/ViewModels/CoinControl/SelectCoinsDialogViewModel.cs

```diff
@@ -43,24 +50,66 @@ public SelectCoinsDialogViewModel(WalletViewModel walletViewModel)
 				pocketColumn
 			}
 		};
+
 		Source.SortBy(pocketColumn, ListSortDirection.Descending);
 		Source.RowSelection!.SingleSelect = true;
 
 		SetupCancel(false, true, false);
 		EnableBack = true;
+		NextCommand = ReactiveCommand.Create(() => Close(DialogResultKind.Normal, GetSelectedCoins(items)));
+	}
+
+	private IEnumerable<SmartCoin> GetSelectedCoins(IEnumerable<CoinControlItemViewModelBase> coinControlItemViewModelBases)
+	{
+		return Flatten(coinControlItemViewModelBases)
+			.OfType<CoinCoinControlItemViewModel>()
+			.Where(x => x.IsSelected == true)
+			.Select(x => x.SmartCoin)
+			.ToList();
 	}
 
 	public HierarchicalTreeDataGridSource<CoinControlItemViewModelBase> Source { get; }
 
+	protected override void OnNavigatedFrom(bool isInHistory)
+	{
+		_disposables.Dispose();
+
+		base.OnNavigatedFrom(isInHistory);
+	}
+
 	private static IColumn<CoinControlItemViewModelBase> ChildrenColumn()
 	{
 		return new HierarchicalExpanderColumn<CoinControlItemViewModelBase>(
-			new PlainTextColumn<CoinControlItemViewModelBase>("", _ => "", GridLength.Auto, null),
+			SelectionColumn(),
 			group => group.Children,
-			node => node.Children.Count > 1,
+			node => node.Children.Count() > 1,
 			node => node.IsExpanded);
 	}
 
+	private static TemplateColumn<CoinControlItemViewModelBase> SelectionColumn()
+	{
+		return new TemplateColumn<CoinControlItemViewModelBase>(
+			"",
+			new FuncDataTemplate<CoinControlItemViewModelBase>(
+				(_, _) => new SelectionCellView(),
+				true),
+			GridLength.Auto);
+	}
+
+	private static void SetSelectionState(IEnumerable<CoinControlItemViewModelBase> items, IEnumerable<SmartCoin> selectedCoins)
+	{
+		var flatten = Flatten(items).ToList();
+		var coins = flatten.OfType<CoinCoinControlItemViewModel>();
+		var coinsToSelect = coins.Select(c => new { c, IsSelected = selectedCoins.Any(other => other  == c.SmartCoin) });
```

Comment:
Overcomplicated, `GetAllCoinChildren` that I mentioned above should be used here too.

Reply:
Done too.